polarization/tissue.py

```diff
@@ -0,0 +1,151 @@
+
+    def scan(self, pulse: Union[Pulse, PulseCollection], verbose=False):
+        if verbose:
+            def v_print(*args, **kwargs):
+                return print(*args, **kwargs)
+        else:
+            v_print = lambda *a, **k: None
+
+        if type(pulse) is Pulse:
+            return self._scanPulse(pulse, v_print)
+        elif type(pulse) is PulseCollection:
+            return self._scanPulseCollection(pulse, v_print)
+        else:
+            return ValueError("Can only scan Pulse or PulseCollection objects. "
+                              "Type {} was provided.".format(type(pulse)))
+
+    def _scanPulse(self, pulse, v_print) -> PulseArray:
+        bScan = []
+        for b in range(self.width):
+            v_print(" .Stack {}/{}".format(b+1, self.width))
+            bScan.append(self.stacks[b].backscatterMany(pulse))
+        return PulseArray(bScan)
+
+    def _scanPulseCollection(self, pulses, v_print) -> PulseCollection:
+        assert not pulses.isExpanded, "Cannot scan a PulseCollection that was already scanned. "
+
+        pulsesOut = []
+        for i, pulse in enumerate(pulses):
+            v_print("Pulse {}/{}".format(i+1, len(pulses)))
+            pulsesOut.append(self._scanPulse(pulse, v_print))
+        return PulseCollection(pulsesOut)
+
+    def __iter__(self):
+        return iter(self.stacks)
+
+    def display(self):
+        """ Display all layer stacks and their properties. """
+        pass
+
+
+class RandomTissue2D(Tissue):
+    def __init__(self, height=3000, width=200,
+                 referenceStack=None, flat=False,
+                 surface=False, maxBirefringence=0.0042, nLayers=None, offset=None, layerHeightRange=(60, 400)):
+
+        if referenceStack is None:
+            referenceStack = RandomTissueStack(surface=surface, maxBirefringence=maxBirefringence,
+                                               nLayers=nLayers, offset=offset, layerHeightRange=layerHeightRange)
+
+        super(RandomTissue2D, self).__init__(referenceStack=referenceStack, height=height, width=width, depth=1)
+
+        self.generateMap(flat)
+        self.generateStacks()
+
+    def generateMap(self, flat=False):
+        initialLengths = [self.referenceStack.offset, *[layer.thickness for layer in self.referenceStack]]
+
+        if not flat:
+            offSets = [RandomSinusGroup(maxA=5, minF=0.001, maxF=0.1, n=40)]
+            offSets.extend([RandomSinusGroup(maxA=2, minF=0.01, maxF=0.1, n=5) for _ in range(self.nLayers)])
+
+            for i, (L, dL) in enumerate(zip(initialLengths, offSets)):
+                self.map[i] = np.array(L + dL.eval(np.arange(self.width)), dtype=int)
+        else:
+            for i, L in enumerate(initialLengths):
+                self.map[i] = np.full(self.width, L, dtype=int)
+
+
+class Sinus:
+    def __init__(self, A=1, f=1, d=0):
+        self.A = A
+        self.f = f
+        self.d = d
+
+    def eval(self, x):
```

Comment:
Hey, look at that:
https://medium.com/swlh/callables-in-python-how-to-make-custom-instance-objects-callable-too-516d6eaf0c8d

you could then do `a = Sinus(x)` and `b=SinusGroups(x)`

```
def __call__(self, x):
         res = 0
         for sinus in self.sinusFunctions:
             res += sinus.eval(x)
         return res
```